server/src/main/java/io/crate/analyze/SwapTableAnalyzer.java

```diff
@@ -68,10 +71,20 @@ public AnalyzedSwapTable analyze(SwapTable<Expression> swapTable,
         }
         SearchPath searchPath = txnCtx.sessionSettings().searchPath();
         Role user = txnCtx.sessionSettings().sessionUser();
-        return new AnalyzedSwapTable(
-            schemas.findRelation(swapTable.source(), Operation.ALTER, user, searchPath),
-            schemas.findRelation(swapTable.target(), Operation.ALTER, user, searchPath),
-            dropSource
-        );
+
+        var source = schemas.findRelation(swapTable.source(), Operation.ALTER_TABLE_RENAME, user, searchPath);
+        if (!(source instanceof DocTableInfo sourceRelation)) {
```

Comment:
The check, in combination with the error message thrown do not cover: blob tables & foreign tables.

Reply:
How does this fix the LR issue?